Sage/carbon
# fix(popover-container): prevent click outside calling onclose when container is not open

Clicking outside the Popover container should call the onClose callback only when the container was previously open and is closed by the click - subsequent clicks should not call the callback.

fix #6066

### Proposed behaviour

The onClose callback should not be called when clicking outside the container if it is not currently open.



### Current behaviour

The onClose callback is called when clicking anywhere on the page other than the button that opens the container.



### Checklist




#### QA


### Additional context



### Testing instructions



The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.


https://codesandbox.io/s/quizzical-tu-2on5yy?file=/src/App.tsx

Test by observing the console output when clicking anywhere on the page (outside the container-open button) when the container is not open.

Ensure the callback still runs as expected when clicking outside to close an open container.
